hydra/utils.py

```diff
@@ -47,31 +50,31 @@ def call(config: Any, *args: Any, **kwargs: Any) -> Any:
         raise HydraException(f"Unsupported config type : {type(config).__name__}")
 
     # make a copy to ensure we do not change the provided object
-    config_copy = OmegaConf.structured(config)
+    config_copy = OmegaConf.structured(config, flags={"allow_objects": True})
     if OmegaConf.is_config(config):
         config_copy._set_parent(config._get_parent())
     config = config_copy
 
     cls = "<unknown>"
     try:
-        assert isinstance(config, DictConfig)
+        assert OmegaConf.is_config(config)
         OmegaConf.set_readonly(config, False)
         OmegaConf.set_struct(config, False)
+        config._set_flag("allow_objects", True)
         cls = _get_cls_name(config)
         type_or_callable = _locate(cls)
```

Comment:
Thanks for supporting objects! Since now any object can be put in the dict, can you allow `_target_` to be the callable object itself?

Reply:
This is potentially doable but out of scope for this diff.
Can you open a new feature request and show an example where this would be preferred over passing the class/function name?